### ISSUE TITLE
Deliver Subject field in Push Consumer Config should not have a default. 

### ISSUE THREAD

**Issue by simon-connektica** [nats-io/nats.rs#1203]
### Proposed change

The `deliver_subject` field is required in order for a push consumer to work properly, at least mine was not working properly without it. The issue is that `pull::Config` implements `Default` and `#[serde(default)]` for the `deliver_subject` field. It's really easy to mess up the configuration such as in the code below: 

```
.get_or_create_consumer(
    cka_nats::subject::worker_logs::LOG_RECORDER_CONSUMER,
    jetstream::push::Config {
        name: "bob"
        durable_name: Some(
            "bob"
        ),
        ..Default::default()
    },
)
```

### Use case

Just making the API more fool proof because it's really easy to miss that this field is required from the docs.

### Contribution

I could do it if the changes is deemed worthwhile.

**Comment by simon-connektica** [comment]
It seem like the Go SDK [creates a new inbox if the DeliverSubject field is empty](https://github.com/nats-io/nats.go/blob/ad97d94c4481d81e72ae125e0821354b5034bf9b/js.go#L1641C17-L1641C25). Going to dig and try to see what async_nats does.

**Comment by Jarema** [comment]
This is the old JS API that uses defaults for everything when calling `Subscribe` method, not if creating a Consumer directly.
I need to think about this one for a bit. I would prefer to avoid "magic" in the background.

At the same time, removing `Default` implementation does not seem sensible, as this is the way to specify only required fields in Consumer Config.

I don't like it, but maybe throwing a runtime error that delivery subject cannot be empty is the way to go here.

**Comment by simon-connektica** [comment]
Returning an error from `get_or_create_consumer` would satisfy me!

**Comment by Jarema** [comment]
@simon-connektica I disagree with the idea of automatically setting the deliver subject and here is the reason:

If someone have strick permissions for subjects, and forgets to setup the delivery subject, this consumer will not work properly.
I prefer the error.

However, there already is an error returned:

```
Error { kind: Other, source: Some(Custom { kind: Other, error: "push consumer must have delivery subject" }) }
```

**Comment by Jarema** [comment]
This is the error returned by the server. What happened when you tried to create a push consumer without a delivery subject?
